lib/lte_link_control/lte_lc.c

```diff
@@ -354,6 +400,16 @@ static void at_handler_ncellmeas(const char *response)
 		return;
 	}
 
+	if (ncellmeas_params.search_type > LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_COMPLETE) {
```

Comment:
How about having a guard that prevents subsequent calls to `lte_lc_neighbor_cell_measurement()` before the initial call has concluded. The API could return `-EINPROGRESS` if there is an ongoing search.

Reply:
Yes, that would be yet another option. Can be done if required. Just wanted to have the functionality similar as it was and imo, code simpler.

But anyways, that is not solving the case where ncellmeas is triggered from at command and not from lte_lc_neighbor_cell_measurement(). See my prev comment.

Reply:
You could make it so that responses that haven't been triggered with `lte_lc_neighbor_cell_measurement()` are ignored. If the link controller is used to perform ncell measurements the provided API should be used.

If a user want to use raw at commands to trigger ncell measurements they can also setup their own at_monitor handler to handle the respons.

Reply:
OK, I did following changes:
- lte_lc_neighbor_cell_measurement() returns -EINPROGRESS if ncellmeas ongoing
- ncellmeas notifs handled in lte_lc only when lte_lc_neighbor_cell_measurement() is on going